lib/src/SortableList/SortableList.tsx

```diff
@@ -0,0 +1,265 @@
+    const [affectedIndex, setAffectedIndex] = React.useState<number>(null);
+    const [affectedNode, setAffectedNode] = React.useState<HTMLDivElement>(null);
+
+    /**
+     * when user clicks on sortable item, set initial delta and selected item
+     * @param event mouse event
+     * @param index selected index
+     */
+    const onMouseDown = (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.TouchEvent<HTMLDivElement>, index: number) => {
+        const parent: HTMLDivElement = (event.target as HTMLDivElement).closest(".sortable-item-wrapper");
+        const position: React.MouseEvent = ((event as React.TouchEvent).touches ? (event as React.TouchEvent).touches[0] : event) as React.MouseEvent;
+        setDelta({
+            x: position.pageX - parent.getBoundingClientRect().left,
+            y: position.pageY - parent.getBoundingClientRect().top,
+        });
+        setCurrentItem(index);
+        setCurrentItemNode(parent);
+    };
+
+    /**
+     * on drag initiated, set the ghost item
+     * @param event drag event
+     */
+    const onDragStart = React.useCallback(
+        (event: React.DragEvent<HTMLDivElement>) => {
+            setIsDragging(() => {
+                const clonedNode: HTMLElement = currentItemNode.cloneNode(true) as HTMLElement;
+                clonedNode.classList.remove("is-active");
+                clonedNode.classList.add("on-drag");
+                clonedNode.style.width = `${currentItemNode.getBoundingClientRect().width}px`;
+                clonedNode.style.height = `${currentItemNode.getBoundingClientRect().height}px`;
+                document.body.appendChild(clonedNode);
+                event.dataTransfer.setDragImage(clonedNode, delta.x, delta.y);
+                event.dataTransfer.effectAllowed = "move";
+                setDragNode(clonedNode);
+                return true;
+            });
+        },
+        [currentItemNode]
+    );
+
+    /**
+     * when the dragged item is dragged over another sortable item, check if the overlapped element fulfills criteria to be swapped
+     * @param event drag event
+     * @param index overlapped index
+     */
+    const onDragOver = React.useCallback(
+        (event: React.DragEvent<HTMLDivElement>, index: number) => {
+            event.preventDefault(); // to prevent ghost image for reverting to original position
+            event.dataTransfer.dropEffect = "move";
+            const activeNodeRect: DOMRect = currentItemNode.getBoundingClientRect();
+            const parent: HTMLDivElement = (event.target as HTMLDivElement).closest(".sortable-item-wrapper");
+            const nodeRect: DOMRect = parent.getBoundingClientRect();
+            const ghostImagePositionTop: number = event.clientY - delta.y;
+            const ghostImagePositionBottom: number = ghostImagePositionTop + activeNodeRect.height; // get ghost image's actual position
+            const isAbove: boolean = nodeRect.top <= ghostImagePositionTop;
```

Comment:
may we can describe explicitly the context of the above flag, something like `isGhostImageAbove{CurrentNode|Node}`